w3af/plugins/crawl/url_fuzzer.py

```diff
@@ -269,6 +314,33 @@ def _mutate_file_type(self, url):
                 url_copy.set_extension(filetype)
                 yield url_copy
 
+    def _mutate_file_name(self, url):
+        filename = url.get_file_name()
+        if filename:
+            domain = url.get_domain_path()
+            url_string = domain.url_string
+            name = filename[:filename.rfind(u'.')]
+            extension = url.get_extension()
+
+            mutate_name_testing = (
+                url_string + '#' + filename + '#',
+                url_string + name + ' (copy).' + extension,
```

Comment:
And also, following the same example above where the filename has no "name", what happens in these lines when `extension` is empty? Does it make sense to send these tests?

These cases should be investigated and handled:
 * http://w3af.org/foobar
 * http://w3af.org/foobar.
 * http://w3af.org/.hidden
 * http://w3af.org/.

Reply:
Yes it wasn't handling these files correctly, I did some modifications and I believe it does it well now.

Here are the tests I did and the results : https://pastebin.com/WG5szsnB (GitHub importing files system doesn't work right now so I put the output on Pastebin)